File: gourmetgate/gourmetgate.core/src/main/java/org/gourmetgate/gourmetgate/core/parameter/BooleanParameter.java
package org.gourmetgate.gourmetgate.core.parameter;

public class BooleanParameter extends AbstractParameter<Boolean> {

  public BooleanParameter(String name) {
    super(name);
  }

  @Override
  protected Boolean parseValue(String value) {
    return Boolean.valueOf(value);
  }

  @Override
  public String prepareValue(Boolean value) {
    return Boolean.toString(value);
  }
}

File: gourmetgate/gourmetgate.ui/src/main/js/parameter/BooleanParameter.ts
import {Parameter} from "./Parameter";

export class BooleanParameter extends Parameter<boolean> {

  protected override _parseValue(stringValue: string): boolean {
    return stringValue.toLowerCase() == 'true'
  }

  protected override _prepareValue(value: boolean): string {
    return value.toString();
  }
}
